python_modules/libraries/dagster-components/dagster_components/component_scaffolding.py

```diff
@@ -17,22 +23,41 @@ def write_line_break(self) -> None:
         super().write_line_break()
 
 
-def scaffold_component_yaml(
+def scaffold_component_decl(
     request: ScaffoldRequest, attributes: Optional[Mapping[str, Any]]
 ) -> None:
-    with open(request.target_path / "component.yaml", "w") as f:
-        component_data = {"type": request.type_name, "attributes": attributes or {}}
-        yaml.dump(
-            component_data, f, Dumper=ComponentDumper, sort_keys=False, default_flow_style=False
+    if request.decl_format == "yaml":
+        with open(request.target_path / "component.yaml", "w") as f:
+            component_data = {"type": request.type_name, "attributes": attributes or {}}
+            yaml.dump(
+                component_data, f, Dumper=ComponentDumper, sort_keys=False, default_flow_style=False
+            )
+            f.writelines([""])
+
+    elif request.decl_format == "python":
+        class_name = request.type_name.split(".")[-1]  # get the class name only
+        package_name = ".".join(request.type_name.split(".")[:-1])  # package name
+        with open(request.target_path / "component.py", "w") as f:
```

Comment:
I would imagine we'd have some utilities akin to `scaffold_component_yaml` for this case